native/cocos/renderer/gfx-metal/MTLDevice.mm

```diff
@@ -193,11 +190,16 @@ of this software and associated engine source code (the "Software"), a limited,
     CC_ASSERT(!_memoryStatus.textureSize); // Texture memory leaked
 }
 
+void CCMTLDevice::frameSync() {
+    constexpr uint8_t TOTAL_FRAME_COUNT = MAX_FRAMES_IN_FLIGHT;
+    while(_inFlightCount.load(std::memory_order_acquire) == TOTAL_FRAME_COUNT) {
+        std::this_thread::sleep_for(std::chrono::milliseconds(10));
+    }
+}
```

Comment:
It might be better to use semaphore to synchronize inflight command buffer.

`_inFlightCount` may be updated just after `load` and before `sleep`. So in the worst case, maybe 10ms is wasted.

Reply:
moved to cmdbuffer, gfx command buffer holds waits when mtlCommandbuffer is submitted(mtlCommandBuffer == nil).